### PR TITLE
cmake: Add option to disable installing PAM files

### PR DESCRIPTION
**Pull request by Conan-Kudo**
This is primarily useful for development purposes.

### CODE REVIEW

**Comment by Conan-Kudo** [comment]
FYI @aleixpol and @davidedmundson, I finally got around to doing this...

File: services/redhat.sddm-autologin.pam [hunk 1]
```diff
@@ -0,0 +1,20 @@
+ #%PAM-1.0
```
Comment (Vogtinator) [inline services/redhat.sddm-autologin.pam]: Whitespace?

File: services/redhat.sddm.pam [hunk 2]
```diff
@@ -0,0 +1,23 @@
+auth     [success=done ignore=ignore default=bad] pam_selinux_permit.so
```
Comment (Vogtinator) [inline services/redhat.sddm.pam]: No `#%PAM-1.0`?

**Review by Vogtinator — CHANGES_REQUESTED** [review]
Whitespace error.

I wonder what the benefit of pam files here in sddm instead of in downstream packaging is. It's just more bureaucracy without much benefit, as upstream reviewers can't really review distro specific changes.

File: services/redhat.sddm-autologin.pam (hunk 1 above)
Comment (Conan-Kudo) [reply services/redhat.sddm-autologin.pam]: Fixed.

File: services/redhat.sddm.pam (hunk 2 above)
Comment (Conan-Kudo) [reply services/redhat.sddm.pam]: Fixed.

**Comment by Conan-Kudo** [comment]
> I wonder what the benefit of pam files here in sddm instead of in downstream packaging is. It's just more bureaucracy without much benefit, as upstream reviewers can't really review distro specific changes.

I was asked to submit them so that folks hacking on SDDM could easily do so on Fedora.

**Comment by Vogtinator** [comment]
> > I wonder what the benefit of pam files here in sddm instead of in downstream packaging is. It's just more bureaucracy without much benefit, as upstream reviewers can't really review distro specific changes.
> 
> I was asked to submit them so that folks hacking on SDDM could easily do so on Fedora.

In that case just not overwriting the distro provided PAM files with `cmake --install` might be the best option. That's how I did it so far, maybe that just needs to be made easier and documented?

**Comment by Conan-Kudo** [comment]
Yeah. I think that would make sense too.

**Comment by Conan-Kudo** [comment]
@Vogtinator I've updated this to add an option to not install PAM files instead.

**Review by Vogtinator — APPROVED** [review]
LGTM. Though I wonder whether the default should be "OFF" or even the PAM services removed completely to avoid mismatches. The current fallback is Arch specific and won't work anywhere else.

**Pull request merged**